server/portal/apps/search/tasks.py

```diff
@@ -26,17 +26,21 @@ def agave_indexer(self, systemId, filePath='/', recurse=True, update_pems=False,
     if not filePath.startswith('/'):
         filePath = '/' + filePath
 
+    if next(sys for sys in settings.PORTAL_DATAFILES_STORAGE_SYSTEMS
+        if sys['name'] == 'Shared Workspaces' and sys['hideSearchBar'] == True
+        and systemId.startswith(settings.PORTAL_PROJECTS_SYSTEM_PREFIX)):
+            return
+
     try:
         filePath, folders, files = walk_levels(client, systemId, filePath, ignore_hidden=ignore_hidden).__next__()
+        index_level(filePath, folders, files, systemId, reindex=reindex)
+        if recurse:
+            for child in folders:
+                self.delay(systemId, filePath=child.path, reindex=reindex)
     except Exception as exc:
         logger.error("Error walking files under system {} and path {}".format(systemId, filePath))
         raise self.retry(exc=exc)
 
```

Comment:
I think we can keep this outside the `try` block so as not to squash exceptions in `self.delay`
```suggestion
        index_level(filePath, folders, files, systemId, reindex=reindex)
    except Exception as exc:
        logger.error("Error walking files under system {} and path {}".format(systemId, filePath))
        raise self.retry(exc=exc)

    if recurse:
        for child in folders:
            self.delay(systemId, filePath=child.path, reindex=reindex)
```